### PR TITLE
Dnf getpackages

### PR DESCRIPTION
**Pull request by Gundersanne**

### CODE REVIEW

File: internal/server/repository.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+func (err *DNFError) Error() string {
+	return fmt.Sprintf("DNF error occured: %s: %s", err.Kind, err.Reason)
+}
+
+type Repo struct {
+	Baseurl    *string `json:"baseurl,omitempty"`
+	Metalink   *string `json:"metalink,omitempty"`
+	Mirrorlist *string `json:"mirrorlist,omitempty"`
+	Rhsm       bool    `json:"rhsm"`
```
Comment (teg) [inline internal/server/repository.go:28]: We need the container to be subscribed now, I guess?

**Review by teg — APPROVED** [review]
Nice! Though do we need some extra magic for RHSM?

File: internal/server/repository.go (hunk 1 above)
Comment (Gundersanne) [reply internal/server/repository.go:28]: ah dang you're right.

File: templates/image-builder.yml [hunk 2]
```diff
@@ -210,3 +212,6 @@ parameters:
   - name: ALLOWED_ORG_IDS
     description: Organization ids allowed to access the api, wildcard means everyone
     value: ""
+  - name: DISTRIBUTIONS_DIR
+    description: Directory which dnf-json uses as a cache
```
Comment (teg) [inline templates/image-builder.yml:216]: Wrong description?

File: dnf-json [hunk 3]
```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/python3
```
Comment (teg) [inline dnf-json:1]: This is now used only to populate the distro files, right? In which case: we should probably document somewhere how to re-run that?

**Review by teg — APPROVED** [review]
Thanks for fixing this up! Happy to merge as-is and address any minor issues in follow-ups!

**Comment by Gundersanne** [comment]
@teg waiting on jenkins currently before merging, agree with all of your comments :)

File: dnf-json (hunk 3 above)
Comment (Gundersanne) [reply dnf-json:1]: Made https://github.com/osbuild/image-builder/issues/126 to track that

**Review by teg — APPROVED** [review]
Perfection!

**Pull request merged**